### PR TITLE
Remove static references to `cl_oo_source`

### LINKED ISSUE #5895: Usage of `cl_oo_source`
abapGit uses class `cl_oo_source`. This class has a predecessor with interface `if_oo_clif_source` and a new implementation starting with release 7.31. This interface is used by abapGit, too.

Since `cl_oo_source` is obsolete in new releases, I want to delete it in one of the upcoming SAP releases. 

I found following usages which are needed in release 7.02:

- https://github.com/abapGit/abapGit/blob/bf5199b8d2284fa7c664884d5c54c239cd0b9210/src/apack/zcl_abapgit_apack_migration.clas.abap#L92
- https://github.com/abapGit/abapGit/blob/c63d92e6f1dafd5565b6009c0da5bcb97195d612/src/objects/oo/zcl_abapgit_oo_serializer.clas.abap#L254
- And one comment ;) https://github.com/abapGit/abapGit/blob/c63d92e6f1dafd5565b6009c0da5bcb97195d612/src/objects/oo/zcl_abapgit_oo_serializer.clas.abap#L175

Can we replace the usages of `cl_oo_source` with dynamic calls?

### PR DESCRIPTION
**Pull request by schneidermic0**
Closes #5895

### CODE REVIEW

File: src/objects/oo/zcl_abapgit_oo_serializer.clas.abap [hunk 1]
```diff
@@ -251,9 +251,9 @@ CLASS zcl_abapgit_oo_serializer IMPLEMENTATION.
 
   METHOD serialize_abap_old.
 * for old ABAP AS versions
-    DATA: lo_source TYPE REF TO cl_oo_source.
+    DATA: lo_source TYPE REF TO object.
 
-    CREATE OBJECT lo_source
+    CREATE OBJECT lo_source TYPE ('CL_OO_SOURCE')
```
Comment (larshp) [inline src/objects/oo/zcl_abapgit_oo_serializer.clas.abap:256]: this is the "serialize_abap_old", which is called when `CALL METHOD ('CL_OO_FACTORY')=>('CREATE_INSTANCE')` fails, so guess this change is okay

Comment (schneidermic0) [reply src/objects/oo/zcl_abapgit_oo_serializer.clas.abap:256]: Yes, I think it is only needed as fallback in 7.02.

**Comment by schneidermic0** [comment]
I didn't find time to test it in 7.02, yet. Therefore, this PR is still a draft.

**Comment by mbtools** [comment]
I tested it in 702. Works

**Comment by schneidermic0** [comment]
@mbtools Thanks for testing. 👍 

I started testing and found #5903.

**Pull request merged**